### PR TITLE
[Enhencement](JDBC Catalog) refactor jdbc catalog insert logic

### PR DESCRIPTION
**Pull request by zy-kkk**
# Proposed changes

Issue Number: close #xxx

## Problem summary

This PR refactors the old way of writing data to JDBC External Table & JDBC Catalog, mainly including the following tasks
1. Continuing the work of @BePPPower 's PR #18594, changing the logic of splicing Inster sql to operating off-heap memory and using preparedStatement.set to write data logic to complete
2. Supplement the support written by largeint type, mainly to adapt to Java.Math.BigInteger, which uses binary operations
3. Delete the splicing SQL logic in the JDBC External Table & JDBC Catalog related written code

ToDo: Binary type，like bit,binary, blob...

Finally, special thanks to @BePPPower , @AshinGau  for his work

Test:
mysql :
image

oracle:
image

postgresql:
image

sqlserver:
image

clickhouse:
image

trino:
image

oceanbase:
image

sap hana:
image
image

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java [hunk 1]
```diff
@@ -154,22 +158,137 @@ public int write(String sql) throws UdfRuntimeException {
...
+            case BIGINT:
+                preparedStatement.setLong(parameterIndex, column.getLong(rowIdx));
+                break;
+            case FLOAT:
+                preparedStatement.setFloat(parameterIndex, column.getFloat(rowIdx));
+                break;
+            case DOUBLE:
+                preparedStatement.setDouble(parameterIndex, column.getDouble(rowIdx));
+                break;
+            case LARGEINT:
```
Comment (AshinGau) [inline fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java:222]: `LARGEINT` should add insertion method in `org.apache.doris.jni.vec.VectorColumn#appendValue`.

File: fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java [hunk 2]
```diff
@@ -154,22 +158,137 @@ public int write(String sql) throws UdfRuntimeException {
...
+            case DECIMALV2:
+            case DECIMAL32:
+            case DECIMAL64:
+            case DECIMAL128:
+                preparedStatement.setBigDecimal(parameterIndex, column.getDecimal(rowIdx));
```
Comment (AshinGau) [inline fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java:227]: `LARGEINT` is the `java.math.BigInteger` in java, so we should add `getBigInt()` in `org.apache.doris.jni.vec.ColumnValue`.

**Comment by AshinGau** [comment]
We'd better add `largeint` test case in `JniScannerTest`, which simulates the memory conversion process of c++ block and java vector table.

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
> We'd better add `largeint` test case in `JniScannerTest`, which simulates the memory conversion process of c++ block and java vector table.

done

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM
Please add regression test in your next PR.

**Comment by AshinGau** [comment]
LGTM

**Review by LemonLiTree — APPROVED** [review]
LGTM

**Review by AshinGau — APPROVED** [review]

**Pull request merged**